agent/engine/engine_sudo_linux_integ_test.go

```diff
@@ -424,6 +424,89 @@ func TestExecCommandAgent(t *testing.T) {
 	os.RemoveAll(execcmd.ECSAgentExecConfigDir)
 }
 
+// TestManagedAgentEvent validates the emitted container events for a started and a stopped managed agent.
+func TestManagedAgentEvent(t *testing.T) {
+	testcases := []struct {
+		Name                 string
+		ExpectedStatus       apicontainerstatus.ManagedAgentStatus
+		ManagedAgentLifetime time.Duration
+		ShouldBeRunning      bool
+	}{
+		{
+			Name:                 "Confirmed emit RUNNING event",
+			ExpectedStatus:       apicontainerstatus.ManagedAgentRunning,
+			ManagedAgentLifetime: 1,
+			ShouldBeRunning:      true,
+		},
+		{
+			Name:                 "Confirmed emit STOPPED event",
+			ExpectedStatus:       apicontainerstatus.ManagedAgentStopped,
+			ManagedAgentLifetime: 0,
+			ShouldBeRunning:      false,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.Name, func(t *testing.T) {
+
+			const (
+				testTaskId        = "exec-command-agent-test-task"
+				testContainerName = "exec-command-agent-test-container"
+			)
+
+			client, err := sdkClient.NewClientWithOpts(sdkClient.WithHost(endpoint), sdkClient.WithVersion(sdkclientfactory.GetDefaultVersion().String()))
+			require.NoError(t, err, "Creating go docker client failed")
+
+			testExecCmdHostBinDir, err := filepath.Abs("../../misc/exec-command-agent-test")
```

Comment:
note: this will need to change after https://github.com/aws/amazon-ecs-agent/pull/2701 gets merged.

See: https://github.com/aws/amazon-ecs-agent/pull/2701/files#diff-8b609ac402c5957da4b973469c270e0047c06a8beefabfaca9555ca0dd61f83dR367

Reply:
I guess if this PR gets merged first I could just change this test in https://github.com/aws/amazon-ecs-agent/pull/2701 before merging